### PR TITLE
Allow passing an optional force option to device reboot and shutdown

### LINKED ISSUE #228: Allow forcing a device reboot or shutdown
From resin-io/resin-supervisor#138, the supervisor's reboot and shutdown endpoints will accept a `force` option as they will start blocking on the update lock. The SDK functions for reboot and shutdown should be changed to accept a `force` option just like `update` does.

### PR DESCRIPTION
**Pull request by pcarranzav**
Closes #228

### CODE REVIEW

**Comment by jviotti** [comment]
Lets regenerate the documentation by running `npm run docs`.

The CI builds seem to be failing for an unrelated download issue. I'll investigate.

File: lib/models/device.coffee [hunk 1]
```diff
@@ -707,6 +707,8 @@ exports.restart = exports.restartApplication
 # @memberof resin.models.device
 #
 # @param {String} uuid - device uuid
+# @param {Object} options - options (optional)
```
Comment (lekkas) [inline lib/models/device.coffee]: nit: you can mark optional arguments using brackets like `[options] - options` in JSDoc

**Comment by pcarranzav** [comment]
@jviotti  The CI error will be fixed with #230

**Comment by jviotti** [comment]
Ah, you fixed it already in https://github.com/resin-io/resin-sdk/pull/230. Awesome!

File: lib/models/device.coffee (hunk 1 above)
Comment (pcarranzav) [inline lib/models/device.coffee]: Done

**Comment by pcarranzav** [comment]
@jviotti regenerated the docs. For some reason running `npm run docs` fails, so I run `jsdoc2md` manually.
This is the error I got:
```
pablo@pablo-laptop:~/resin/resin-sdk$ npm run docs

> resin-sdk@5.3.6 docs /home/pablo/resin/resin-sdk
> jsdoc2md build/{,**/}*.js > DOCUMENTATION.md

/usr/bin/env: node --no-warnings: No such file or directory

npm ERR! Linux 4.3.0-040300-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "docs"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! resin-sdk@5.3.6 docs: `jsdoc2md build/{,**/}*.js > DOCUMENTATION.md`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the resin-sdk@5.3.6 docs script 'jsdoc2md build/{,**/}*.js > DOCUMENTATION.md'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the resin-sdk package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     jsdoc2md build/{,**/}*.js > DOCUMENTATION.md
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs resin-sdk
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls resin-sdk
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/pablo/resin/resin-sdk/npm-debug.log
```
Worth filing an issue? It looks like it happens whenever I try to run jsdoc2md from node_modules instead of /usr/bin.

**Comment by jviotti** [comment]
@pcarranzav That's really weird:

```
/usr/bin/env: node --no-warnings: No such file or directory
```

Is the NodeJS binary called `node` or `nodejs` in your system? What happens if you globally install `jsdoc2md` globally first? 

If you can open an issue about it, that'd be great, otherwise happy to generate the docs for you.

**Comment by jviotti** [comment]
Ah, so you generated them manually already. Awesome!

**Comment by pcarranzav** [comment]
@jviotti the docs error is fixed with #231 
Do you approve this one (i.e. #229)?

**Comment by jviotti** [comment]
LGTM :+1:

**Comment by pcarranzav** [comment]
@jviotti is there a way to retry the appveyor tests? looks like they failed due to a (probably transient) network issue?

**Comment by pcarranzav** [comment]
Oh, just found out how to do it myself

**Comment by pcarranzav** [comment]
@jviotti would you please publish this so that I include it in resin-io/resin-cli#404 ?

**Comment by lekkas** [comment]
@pcarranzav if there's a way to restart failed appveyor tests other than updating the PR, please do tell how, I've also stumbled there ! 😄

**Comment by jviotti** [comment]
@pcarranzav I'll publish in a second!

**Comment by jviotti** [comment]
Checkout v5.4.0 on NPM!

**Comment by pcarranzav** [comment]
Awesome, thanks @jviotti !

**Pull request merged**